extension/memorylimiterextension/memorylimiter.go

```diff
@@ -0,0 +1,40 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package memorylimiterextension // import "go.opentelemetry.io/collector/extension/memorylimiterextension"
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/internal/memorylimiter"
+)
+
+// newMemoryLimiter returns a new memorylimiter extension.
+func newMemoryLimiter(cfg *Config, logger *zap.Logger) (*memoryLimiterExtension, error) {
+	ml, err := memorylimiter.NewMemoryLimiter(cfg, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return &memoryLimiterExtension{memLimiter: ml}, nil
+}
+
+func (ml *memoryLimiterExtension) Start(ctx context.Context, host component.Host) error {
+	return ml.memLimiter.Start(ctx, host)
+}
+
+func (ml *memoryLimiterExtension) Shutdown(ctx context.Context) error {
+	return ml.memLimiter.Shutdown(ctx)
+}
+
+// MustRefuse returns if the caller should deny because memory has reached it's configured limits
+func (ml *memoryLimiterExtension) MustRefuse() bool {
+	return ml.memLimiter.MustRefuse()
+}
+
+func (ml *memoryLimiterExtension) Dependencies() []component.ID {
+	return []component.ID{component.NewID("memory_ballast")}
+}
```

Comment:
As I mentioned in https://github.com/open-telemetry/opentelemetry-collector/pull/8964/files#r1441192372, please remove this. Sorry for the back-and-forths.

Reply:
no worries

Reply:
removed